qoqo-qryd/tests/integration/experimental_devices.rs

```diff
@@ -0,0 +1,605 @@
+        device_mut
+            .call_method1("add_qubit_tweezer_mapping", (1, 2))
+            .unwrap();
+        device_mut
+            .call_method1("add_qubit_tweezer_mapping", (2, 3))
+            .unwrap();
+        device_mut
+            .call_method1("add_qubit_tweezer_mapping", (3, 0))
+            .unwrap();
+
+        assert!(device
+            .call_method1("switch_layout", ("OtherLayout",))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("switch_layout", ("OtherLayout",))
+            .is_ok());
+
+        assert!(device
+            .call_method1("single_qubit_gate_time", ("PauliX", 3))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("single_qubit_gate_time", ("PauliX", 3))
+            .is_ok());
+        assert!(device
+            .call_method1("two_qubit_gate_time", ("CNOT", 3, 0))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("two_qubit_gate_time", ("CNOT", 3, 0))
+            .is_ok());
+        assert!(device
+            .call_method1("three_qubit_gate_time", ("Toffoli", 3, 0, 1))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("three_qubit_gate_time", ("Toffoli", 3, 0, 1))
+            .is_ok());
+        assert!(device
+            .call_method1("multi_qubit_gate_time", ("MultiQubitZZ", vec![3, 0, 1, 2]))
+            .is_ok());
+        assert!(device_mut
+            .call_method1("multi_qubit_gate_time", ("MultiQubitZZ", vec![3, 0, 1, 2]))
+            .is_ok());
+    })
+}
+
+/// Test number_qubits function of ExperimentalDeviceWrapper and ExperimentalMutableDeviceWrapper
+#[test]
+fn test_number_qubits() {
+    // Setup fake preconfigured device
+    let mut exp = ExperimentalDevice::new();
+    exp.set_tweezer_single_qubit_gate_time("PauliX", 0, 0.23, None);
+    exp.set_tweezer_single_qubit_gate_time("PauliX", 1, 0.23, None);
+    let fake_api_device = ExperimentalDeviceWrapper { internal: exp };
+    pyo3::prepare_freethreaded_python();
+    Python::with_gil(|py| {
+        let fake_api_pypyany = fake_api_device.into_py(py);
+        let device_type_mut = py.get_type::<ExperimentalMutableDeviceWrapper>();
+        let device = fake_api_pypyany.as_ref(py);
+        let device_mut = device_type_mut.call0().unwrap();
+
+        device_mut
+            .call_method1("set_tweezer_single_qubit_gate_time", ("PauliX", 0, 0.23))
+            .unwrap();
+        device_mut
+            .call_method1("set_tweezer_single_qubit_gate_time", ("PauliX", 1, 0.23))
+            .unwrap();
+
+        assert_eq!(
+            device
+                .call_method0("number_qubits")
+                .unwrap()
+                .extract::<usize>()
+                .unwrap(),
+            0
+        );
+        assert_eq!(
+            device_mut
+                .call_method0("number_qubits")
+                .unwrap()
+                .extract::<usize>()
+                .unwrap(),
+            0
+        );
```

Comment:
This should already be two here if we "rebuild" the naive fully occupied qubit to tweezer mapping after `set_tweezer_single_qubit_gate_time`.
Or rather since we revert the mapping to None, we should have a test in number_qubits, if the mapping is None, we just take the max number of tweezers in the stored tweezer gate times.